sklearn/datasets/tests/test_20news.py

```diff
@@ -88,3 +91,27 @@ def test_20news_normalization(fetch_20newsgroups_vectorized_fxt):
 
     assert_allclose_dense_sparse(X_norm, normalize(X))
     assert np.allclose(np.linalg.norm(X_norm.todense(), axis=1), 1)
+
+
+def test_20news_asframe(fetch_20newsgroups_vectorized_fxt):
+    pd = pytest.importorskip('pandas')
+
+    try:
+        bunch = fetch_20newsgroups_vectorized_fxt(as_frame=True)
+    except ValueError as err:
+        assert LooseVersion(pd.__version__) < '0.25'
+        assert "DataFrame requires Pandas v0.25+." in str(err.args[0])
+    else:
+        frame = bunch.frame
+
+        assert frame.shape == (11314, 130108)
+        assert isinstance(bunch.data, pd.DataFrame)
```

Comment:
So we should have a common test now for the type part. Could you only check the shape (and the sparse because this is really specific to this fetcher).